pennylane_lightning/lightning_qubit.py

```diff
@@ -203,7 +203,11 @@ def apply_lightning(self, state, operations, dtype=np.complex128):
         else:
             raise TypeError(f"Unsupported complex Type: {dtype}")
 
-        for o in operations:
+        skipped_ops = ["Identity"]
+        ops_iter = operations if isinstance(operations, Iterable) else [operations]
+        ops_generator = (o for o in ops_iter if o not in skipped_ops)
+
+        for o in ops_generator:
```

Comment:
```suggestion
        skipped_ops = ["Identity"]
        for o in operations:
            if op in skipped_ops:
                continue
```

Thought this would've to be done in C++. Nice to know this can just be done from Python.

This might be a more straightforward way to skip over the `"skipped_ops"`, but the way here would work as well.

Reply:
Happy with either approach on my side.
Though, I guess we should keep that `skipped_ops` list small to avoiding adding an additional `o(n)` comparisons when applying each gate.